### ISSUE TITLE
[zlib] Build error on x86-windows (issue downloading libraries from github?)

### ISSUE THREAD

**Issue by AntonRevin** [microsoft/vcpkg#40746]
Package: zlib:x86-windows@1.3.1

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.40.33814.0
-    vcpkg-tool version: 2024-08-01-fd884a0d390d12783076341bd43d77c3a6a15658
    vcpkg-scripts version: 85392b146 2024-08-30 (27 hours ago)

**To Reproduce**

`vcpkg install `

**Failure logs**

```
-- Downloading https://github.com/madler/zlib/archive/v1.3.1.tar.gz -> madler-zlib-v1.3.1.tar.gz...

CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
      
      Failed to download file with error: no such file or directory
      If you are using a proxy, please check your proxy setting. Possible causes are:
      
      1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
         to `https://address:port`. This is not correct, because `https://` prefix
         claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
         , etc..) is an HTTP proxy. Try setting `http://address:port` to both
         HTTP_PROXY and HTTPS_PROXY instead.
      
      2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
         set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
         The value set by your proxy might be wrong, or have same `https://` prefix issue.
      
      3. Your proxy's remote server is out of service.
      
      If you've tried directly download the link, and believe this is not a temporary
      download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
      to report this upstream download server failure.
      

Call Stack (most recent call first):
  scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
  scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
  ports/zlib/portfile.cmake:2 (vcpkg_from_github)
  scripts/ports.cmake:192 (include)



```

**Additional context**

<details><summary>vcpkg.json</summary>

```
{
  "$schema": "https://raw.githubusercontent.com/microsoft/vcpkg/master/scripts/vcpkg.schema.json",
  "name": "project-name",
  "version-semver": "0.0.0",
  "dependencies": [
    "libtcod",
    "sdl2",
    {
      "name": "sdl2",
      "default-features": false,
      "platform": "windows"
    }
  ]
}

```
</details>

**Comment by autoantwort** [comment]
Hm cant repro locally. Can you open https://github.com/madler/zlib/archive/v1.3.1.tar.gz in your browser? 
Maybe your antivirus software deleted the file after it was downloaded (` no such file or directory`)

**Comment by AntonRevin** [comment]
I am able to access the lib from a browser without issues. Interestingly, I built another project with a similar set of dependencies recently without issue.

**Comment by HunterZ** [comment]
Same issue under MSYS2 MinGW64 on a local Windows 11 machine with no proxy server:
```
[cmake] Installing 1/14 zlib:x64-mingw-dynamic@1.3.1...
[cmake] Building zlib:x64-mingw-dynamic@1.3.1...
[cmake] C:\msys64\opt\vcpkg\triplets\community\x64-mingw-dynamic.cmake: info: loaded community triplet from here. Community triplets are not built in the curated registry and are thus less likely to succeed.
[cmake] -- Downloading https://github.com/madler/zlib/archive/v1.3.1.tar.gz -> madler-zlib-v1.3.1.tar.gz...
[cmake] [DEBUG] To include the environment variables in debug output, pass --debug-env
[cmake] [DEBUG] Trying to load bundleconfig from C:\msys64\opt\vcpkg\vcpkg-bundle.json
[cmake] [DEBUG] Failed to open: C:\msys64\opt\vcpkg\vcpkg-bundle.json
[cmake] [DEBUG] Bundle config: readonly=false, usegitregistry=false, embeddedsha=nullopt, deployment=Git, vsversion=nullopt
[cmake] [DEBUG] Metrics enabled.
[cmake] [DEBUG] Feature flag 'binarycaching' unset
[cmake] [DEBUG] Feature flag 'compilertracking' unset
[cmake] [DEBUG] Feature flag 'registries' unset
[cmake] [DEBUG] Feature flag 'versions' unset
[cmake] [DEBUG] Feature flag 'dependencygraph' unset
[cmake] warning: Download failed -- retrying after 1000ms
[cmake] warning: Download failed -- retrying after 2000ms
[cmake] warning: Download failed -- retrying after 4000ms
[cmake] error: Missing madler-zlib-v1.3.1.tar.gz and downloads are blocked by x-block-origin.
[cmake] error: https://github.com/madler/zlib/archive/v1.3.1.tar.gz: WinHttpSendRequest failed with exit code 10106
[cmake] error: https://github.com/madler/zlib/archive/v1.3.1.tar.gz: WinHttpSendRequest failed with exit code 10106
[cmake] error: https://github.com/madler/zlib/archive/v1.3.1.tar.gz: WinHttpSendRequest failed with exit code 10106
[cmake] error: https://github.com/madler/zlib/archive/v1.3.1.tar.gz: WinHttpSendRequest failed with exit code 10106
[cmake] [DEBUG] D:\a\_work\1\s\src\vcpkg\base\downloads.cpp(1030): 
[cmake] [DEBUG] Time in subprocesses: 0us
[cmake] [DEBUG] Time in parsing JSON: 7us
[cmake] [DEBUG] Time in JSON reader: 0us
[cmake] [DEBUG] Time in filesystem: 1102us
[cmake] [DEBUG] Time in loading ports: 0us
[cmake] [DEBUG] Exiting after 7 s (7027229us)
[cmake] 
[cmake] CMake Error at scripts/cmake/vcpkg_download_distfile.cmake:32 (message):
[cmake]       
[cmake]       Failed to download file with error: 1
[cmake]       If you are using a proxy, please check your proxy setting. Possible causes are:
[cmake]       
[cmake]       1. You are actually using an HTTP proxy, but setting HTTPS_PROXY variable
[cmake]          to `https://address:port`. This is not correct, because `https://` prefix
[cmake]          claims the proxy is an HTTPS proxy, while your proxy (v2ray, shadowsocksr
[cmake]          , etc..) is an HTTP proxy. Try setting `http://address:port` to both
[cmake]          HTTP_PROXY and HTTPS_PROXY instead.
[cmake]       
[cmake]       2. If you are using Windows, vcpkg will automatically use your Windows IE Proxy Settings
[cmake]          set by your proxy software. See https://github.com/microsoft/vcpkg-tool/pull/77
[cmake]          The value set by your proxy might be wrong, or have same `https://` prefix issue.
[cmake]       
[cmake]       3. Your proxy's remote server is out of service.
[cmake]       
[cmake]       If you've tried directly download the link, and believe this is not a temporary
[cmake]       download server failure, please submit an issue at https://github.com/Microsoft/vcpkg/issues
[cmake]       to report this upstream download server failure.
[cmake]       
[cmake] 
[cmake] Call Stack (most recent call first):
[cmake]   scripts/cmake/vcpkg_download_distfile.cmake:270 (z_vcpkg_download_distfile_show_proxy_and_fail)
[cmake]   scripts/cmake/vcpkg_from_github.cmake:106 (vcpkg_download_distfile)
[cmake]   ports/zlib/portfile.cmake:2 (vcpkg_from_github)
[cmake]   scripts/ports.cmake:192 (include)
[cmake] 
[cmake] 
[cmake] error: building zlib:x64-mingw-dynamic failed with: BUILD_FAILED
[cmake] See https://learn.microsoft.com/vcpkg/troubleshoot/build-failures?WT.mc_id=vcpkg_inproduct_cli for more information.
[cmake] Elapsed time to handle zlib:x64-mingw-dynamic: 7.1 s
```

**Comment by dg0yt** [comment]
> [cmake] error: https://github.com/madler/zlib/archive/v1.3.1.tar.gz: WinHttpSendRequest failed with exit code 10106

This is a new issue.

**Comment by HunterZ** [comment]
> This is an automated message. Per our repo policy, stale issues get closed if there has been no activity in the past 28 days. The issue will be automatically closed in 14 days. If you wish to keep this issue open, please add a new comment.

😩

**Comment by FrankXie05** [comment]
@HunterZ Could you please access it manually? https://github.com/madler/zlib/archive/v1.3.1.tar.gz
